src/libostree/ostree-fetcher.c

```diff
@@ -1110,10 +1081,8 @@ on_request_sent (GObject        *object,
                 goto out;
 
               (void) g_input_stream_close (pending->request_body, NULL, NULL);
-              g_queue_insert_sorted (&pending->thread_closure->pending_queue,
-                                     g_object_ref (task), pending_task_compare,
-                                     NULL);
-              remove_pending_rerun_queue (pending);
```

Comment:
We can probably delete this function now and remove the pending object right at the call sites. (Or at least, rename it to just `remove_pending`).